src/AutoMapper/Mappers/StringDictionaryMapper.cs

```diff
@@ -55,8 +55,8 @@ public bool IsMatch(TypePair context)
 
         private static TDestination Map<TDestination>(StringDictionary source, TDestination destination, ResolutionContext context)
         {
-            destination = destination == null ? context.Mapper.CreateObject<TDestination>() : destination;
-            var destTypeDetails = new TypeDetails(typeof(TDestination), _ => true, _ => true);
+            destination = destination == null ? context.Mapper.CreateObject<TDestination>() : destination;
+            var destTypeDetails = new TypeDetails(typeof(TDestination), context.ConfigurationProvider.ConfigurationExpression);
```

Comment:
Why can't we just use context.ConfigurationProvider here? I was hesitant to expose the configuration expression publicly, since that would be effectively mutating config.